std/container/array.d

```diff
@@ -24,6 +24,25 @@ import std.traits;
 
 public import std.container.util;
 
+pure @system unittest
+{
+    enum size_t n = 100;
+    float[] a0 = new float [n];
+    {
+        import std.range : iota;
+        import std.algorithm.mutation : copy;
+        copy(iota(0, n), a0);
+        enum float factor = 1.1f;
+        a0[] *= factor;
+        assert(a0[$ - 1] == (n - 1) * factor);
+    }
+    auto a1 = Array!float(a0);
+    assert(a1.length == n);
+    assert(a1.capacity == n); // Checks that a1's constructor has called reserve.
+    import std.algorithm.comparison : equal;
+    assert(equal(a1[], a0[]));
```

Comment:
Try to make unittests focussed and self-contained if you can. If it's only testing that "a1's constructor has called reserve" then the complex initialization of the float array is not needed:
```D
immutable int[4] a0 = [10, 20, 30, 40];
auto a1 = Array!int(a0[]);
assert(a1.length == a0.length);
assert(a1.capacity == a0.length); // Checks that a1's constructor has called reserve.
assert(a1[] == a0[]);
```

Reply:
> Try to make unittests focussed and self-contained if you can. If it's only testing that "a1's constructor has called reserve" then the complex initialization of the float array is not needed:
> 
> ```d
> immutable int[4] a0 = [10, 20, 30, 40];
> auto a1 = Array!int(a0[]);
> assert(a1.length == a0.length);
> assert(a1.capacity == a0.length); // Checks that a1's constructor has called reserve.
> assert(a1[] == a0[]);
> ```

Unfortunately, for some lengths, the test passes even if reserve is not called (at least with the insertBack version, which is a correct version, even though it might be less efficient than the emplace version).

I have now added even more comments in the source code for the `unittest`, so that others do not fall in the same traps I have.

While the line count has increased, IMO it is quite readable by humans. And (I dare say) useful.